labdao/plex
# Admin attribute and middleware

## What type of PR is this? 

- 🎮 Feature

## Description

This PR introduces a `User` admin attribute and middleware applied on specific routes: `POST /tools` and `PUT /tools/{cid}`. The admin attribute may be used to check if a user should be charged credits for submitting GPU jobs.

## Related Tickets & Documents

#877 - stale

## Steps to Test

1. Login as user
2. Add a new tool. Expected behavior is to get a 401 due to lack of admin status
3. Update admin status of your user in DB
4. Re-attempt to add tool. This should work now 
5. Hit `PUT /tools/{cid-of-your-tool}`. Confirm tool has been updated
6. Logout and login as another, non-admin user
7. Attempt to hit `PUT /tools/{cid-of-your-tool}`. Expected behavior is to receive a 401

## Relevant GIF 

<img src="https://media2.giphy.com/media/8WI3UEK8kXLQ9SHnTx/giphy.gif"/>
